src/hotspot/share/interpreter/oopMapCache.cpp

```diff
@@ -177,9 +174,15 @@ class VerifyClosure : public OffsetClosure {
 
 InterpreterOopMap::InterpreterOopMap() {
   initialize();
-#ifdef ASSERT
-  _resource_allocate_bit_mask = true;
-#endif
+  DEBUG_ONLY(_used = false;)
+}
+
+InterpreterOopMap::~InterpreterOopMap() {
+  if (mask_size() > small_mask_limit) {
+    assert(!Thread::current()->resource_area()->contains((void*)_bit_mask[0]),
+           "The bit mask should be allocated from the C heap");
```

Comment:
Arguably, this assert is not needed. In debug builds, we have NMT enabled, and that does a check on os::free.

However, an assert that _bit_mask[0] != 0 would make sense, since the free quielty swallows null pointers.

Reply:
Fixed. We could had such case if the mask was never filled due to invalid bci, so I also improved the conditional.